### ISSUE TITLE
Add explicit scope to OpenMP crit sec in wf3cte.c

### ISSUE THREAD

**Issue by jamienoss** [spacetelescope/hstcal#362]
[src](https://github.com/spacetelescope/hstcal/blob/master/pkg/wfc3/calwf3/wf3cte/wf3cte.c#L1359-L1364)
```c
#pragma omp critical (cte)
        for (j=0; j< RAZ_ROWS; j++){
            if (Pix(rz.dq.data,i,j)){
                Pix(rc.sci.data,i,j)= pix_modl[j];
            }
        }
```

should be:

```c
#pragma omp critical (cte)
{ // <-----ADD
        for (j=0; j< RAZ_ROWS; j++){
            if (Pix(rz.dq.data,i,j)){
                Pix(rc.sci.data,i,j)= pix_modl[j];
            }
        }
}  // <-----ADD
```

I tried looking through the manual and couldn't find anything concrete that the ``for`` scope would not suffice. Neither in the [4.5 std, sec 2.13.2](https://www.openmp.org/wp-content/uploads/openmp-4.5.pdf) nor the [2.0 std, sec 2.6.2](https://www.openmp.org/wp-content/uploads/cspec20.pdf).

~~I'm not sure if this is or isn't an actual bug~~ (it isn't a bug, but only because the crit-sec isn't even needed - it may be a bug if it were, I don't know). If not, we should still fix for clarity and not having to revisit this question.

@mdlpstsci & @stscicrawford ~~it may be worth testing this as a sanity check for #358.~~ _c.f._ https://github.com/spacetelescope/hstcal/issues/362#issuecomment-427366616

**Comment by jamienoss** [comment]
Actually, now that I just fixed it, I remembered why I hadn't opened this before... the crit sec isn't even needed. In fact, I removed it for the new ACS implementation ([src](https://github.com/spacetelescope/hstcal/blob/master/ctegen2/ctegen2.c#L257-L263)).

The mem access isn't thread-overlapping since ``i`` is the iter var for the actual parallel ``for``. Any argument for having this critical section is also one for sectioning the entire core CTE computation (which un-parallelizes the whole thing, so _can't_ be needed).
